README.md

```diff
@@ -1,53 +1,112 @@
 - Use `yarn` to install project dependencies.
 
-#### Database setup (MariaDB)
+## Start app
 
+Run `yarn start` to start server in production mode.
 
+Run `yarn start:dev` to start with nodemon and development environment.
+### Running in docker:
 
-#### Start app
+With docker-compose:
+```bash
+docker-compose up
+```
 
-Run `yarn start` to start server in production mode.
+## Testing
+### Unit testing
+
+You can run unit tests with:
 
-Run `yarn run dev` to start with nodemon and development environment.
+```bash
+yarn run test
+```
 
-# Running:
+### End to End Testing
 
-With docker-compose:
+Running e2e test requires creating a database first (check .env.test file), and have mariadb started.
 
-`docker-compose up`
+You can either run :
 
-Checkout localhost:3000/api to see every endpoint of the app along documentation
+```bash
+yarn run test:e2e
+```
 
-Locally:
 
-Make sure to have mariadb running in port 6603 and run: `yarn start:dev`
+## Guideline Nest.js
+This project is based in <a href="http://nestjs.com/" target="blank">NestJS</a> and implement DDD (Domain Driven Design).
+Important in DDD is separate this layers:
+- Use cases
+- Persistence
+- Domain
+- Controllers
 
-# Unit testing
+So, following best practices in NestJS we have 2 primary folders:
+- Modules
+- Externals
 
-You can run unit tests with:
+### Modules
+In this folder we have all "use cases" or "context" and following nest.js modules structure, we have 3 concepts of files:
+- <strong>Module:</strong> files with `*.module.ts`
+- <strong>Providers:</strong> files with `*.service.ts`
+- <strong>Controllers:</strong> files with `*.controller.ts`
+- <strong>Model:</strong> files with `*.model.ts`
 
-`yarn run test`
+And following DDD we include this concepts:
+- <strong>Domain:</strong> files with `*.domain.ts`
+- <strong>Repository:</strong> files with `*.repository.ts`
 
-# End to End Testing
+So, example of module "File" `src/modules/file`:
+- <strong>Controllers:</strong> `file.controller.ts` http Requests with endpoints: POST /file/:id/move
+- <strong>Services:</strong> `file.service.ts` use cases of files: moveFile, call Repository to move file in database, and work with domains.
+- <strong>Domain:</strong> `file.domain.ts` Class File, all attributes and business-logic is here, but in this class we not include nothing about persistence or use cases.
+- <strong>Model:</strong> `file.model.ts` Entity Model in Sequelize(MariaDB)
+- <strong>Repository:</strong> `file.repository.ts` Class with all querys using model, but important, Repository response with domains(File) but never with model directly.
+- <strong>Module:</strong> `file.module.ts` Module of Nest.js
 
-Running e2e test requires creating a database first (check .env.test file)
+### Externals
+In this folder we have all librarys of third partys or not business-based.
```

Comment:
"This folder contains third-party dependencies and external services whose usage could be necessary but is not business-related"